provisioning/transport/amqp/src/SaslTpmHandler.cs

```diff
@@ -14,36 +14,35 @@ public class SaslTpmHandler : SaslHandler
     {
         static readonly byte[] EmptyByte = { 0 };
         private const string MechanismName = "TPM";
-        private const int ChallengeKeySegmentSize = 400;
-        private readonly StringBuilder _encodedNonceStringBuilder = new StringBuilder(ChallengeKeySegmentSize * 3);
 
         private readonly byte[] _endorsementKey;
 
-        private readonly string _hostName;
+        private readonly string _idScope;
         private readonly ProvisioningSecurityClientSasToken _security;
         private readonly byte[] _storageRootKey;
+        private byte[] _nonceBuffer = Array.Empty<byte>();
         private byte _nextSequenceNumber;
 
-        public SaslTpmHandler(byte[] endorsementKey, byte[] storageRootKey, string hostName,
+        public SaslTpmHandler(byte[] endorsementKey, byte[] storageRootKey, string idScope, 
             ProvisioningSecurityClientSasToken security)
-
         {
             //TODO: check inputs
 
             Mechanism = MechanismName;
             _endorsementKey = endorsementKey;
             _storageRootKey = storageRootKey;
-            _hostName = hostName;
+            _idScope = idScope;
             _security = security;
         }
 
+        private string HostName => $"{this._idScope}/registrations/{_security.RegistrationID}";
```

Comment:
I think this doesn't look like a hostname anymore. It should be renamed (`_saslScope` maybe?)
Also: please follow the naming conventions and the placing of member declarations (at the top, next to the other private members).

Reply:
it is the value assigned to init.hostname, I will keep the same name (_hostname).

---
In reply to: [148645292](https://github.com/Azure/azure-iot-sdk-csharp/pull/214#discussion_r148645292) [](ancestors = 148645292)